components/match2/wikis/hearthstone/match_summary.lua

```diff
@@ -90,19 +99,11 @@ function CustomMatchSummary._createSubmatch(game)
 	---@param opponentIndex any
 	---@return Html
 	local createScore = function(opponentIndex)
-		local isWinner = opponentIndex == game.winner or game.resultType == 'draw'
-		if game.resultType == 'default' then
-			return OpponentDisplay.BlockScore{
+		local isWinner = opponentIndex == submatch.winner
+		return OpponentDisplay.BlockScore{
 				isWinner = isWinner,
-				scoreText = isWinner and 'W' or string.upper(game.walkover),
+				scoreText = (submatch.scores or {})[opponentIndex],
```

Comment:
with the above changes you can now just use the same as we use on match level

Reply:
Couldn't we use opponent.score ?

Reply:
you want to display score if opponent.status is score status
else you want to display the opponent.status

Reply:
yup